nautobot_device_lifecycle_mgmt/metrics.py

```diff
@@ -0,0 +1,57 @@
+"""Nautobot Device LCM  plugin application level metrics ."""
+from datetime import datetime
+
+from django.conf import settings
+from nautobot.dcim.models import Device, Site, InventoryItem
+from prometheus_client.core import GaugeMetricFamily
+
+from nautobot_device_lifecycle_mgmt.models import HardwareLCM
+
+PLUGIN_SETTINGS = settings.PLUGINS_CONFIG.get("nautobot_device_lifecycle_mgmt", {})
+
+
+def nautobot_metrics_dlcm_eos():
+    """Calculate number of EOS devices per Device Type and per Site.
+    Yields:
+        GaugeMetricFamily: Prometheus Metrics
+    """
+    current_dt = datetime.now()
+    hw_eos_notices = HardwareLCM.objects.filter(end_of_support__lte=current_dt)
+    hw_eos_device_types = [notice.device_type for notice in hw_eos_notices]
+    hw_eos_inventoryitems = [notice.inventory_item for notice in hw_eos_notices]
+
+    part_number_gauge = GaugeMetricFamily(
+        "nautobot_lcm_devices_eos_per_part_number", "Nautobot LCM Devices EOS per Part Number", labels=["part_number"]
+    )
+    devices_gauge = GaugeMetricFamily(
+        "nautobot_lcm_devices_eos_per_site", "Nautobot LCM Devices EOS per Site", labels=["site"]
+    )
+
+    for notice in hw_eos_notices:
+        if notice.device_type:
+            part_number = notice.device_type.part_number
```

Comment:
```suggestion
            part_number = notice.device_type.part_number if notice.device_type.part_number else notice.device_type.slug
```
What if a Device Type has no Part Number entered (which is optional attribute)?

Reply:
true, SKUs are important but not everybody adds them to the SoT. + to commit!